### ISSUE TITLE
DeploymentPrefix without the Dash

### ISSUE THREAD

**Issue by spencerkittleson** [LambdaSharp/LambdaSharpTool#132]
`DeploymentPrefix` and `DeploymentPrefixLowercase` without dashes at the end.  Maybe `Deployment` and `DeploymentLowercase`

Use case Module.yml
```yml
  - Resource: ServiceParameter
    Description: example service parameter
    Type: AWS::SSM::Parameter
    Properties:
      Name: !Sub "/${DeploymentLowercase}/service/property"
      Type: String
      Value: "foo"
```

Workaround has been the following:
```yml
 - Variable: Tier
    Value: !Select [ "0", !Split [ '-' , !Sub '${DeploymentPrefixLowercase}' ] ]
```

**Comment by bjorg** [comment]
Or you can append a value so you don't have a trailing `-`.
!Sub "/${DeploymentLowercase}tier/service/property"

If CFN had string operations for lowercase and append, I would happily oblige. However, dedicated 4 parameters to different variations of the deployment name is excessive, especially since there is a limit of 60 max parameters for a template.

**Comment by bjorg** [comment]
Also, beware that `DeploymentPrefix` can be an empty string when no tier name exists.

**Comment by bjorg** [comment]
Better late than never: in 0.8.0.6, you'll have `Deployment::Tier`,  `Deployment::TierPrefix`,  `Deployment::TierLowercase`, and  `Deployment::TierPrefixLowercase` to play with!

**Comment by bjorg** [comment]
This is fixed in v0.8.0.6.
